### PR TITLE
[snap/frontend]: Added transfer form component

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- The transfer transaction form is missing when clicking on the send button on the home page.

## What's the fix?
- Added transfer transaction form.
- Click on the **Send** button to pop up the modal box.
- Added form validation, such as Address (network check), mosaic amount, and message length.
- Handle mosaic selector logic, when the user clicks on "Add" mosaic, the options will be filtered (no duplicate).

Preview:

image

### CODE REVIEW

File: snap/frontend/components/TransferModalBox/index.jsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+	const validate = (field, value) => {
+		switch (field) {
+		case 'address':
+			return facade.network.isValidAddressString(value) ? null : 'Invalid address';
+		case 'message':
+			return 1024 >= value.length ? null : 'Message length should not exceed 1024 characters';
+		case 'amount':
+			return value.map((amount, index) => {
+				const maxBalance = mosaicsBalance.find(mosaic => mosaic.id === selectedMosaics[index].id).amount;
```
Comment (Wayonb) [inline snap/frontend/components/TransferModalBox/index.jsx:33]: maxBalance => currentBalance?

File: snap/frontend/components/TransferModalBox/index.jsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+				return absoluteAmount > maxBalance ? 'Not enough balance' : null;
+			});
+		default:
+			return null;
+		}
+	};
+
+	const validateForm = () => {
+		const validate = (field, value) => {
```
Comment (Wayonb) [inline snap/frontend/components/TransferModalBox/index.jsx:44]: This validate function looks the same as above function?

**Review by Wayonb — CHANGES_REQUESTED** [review]
It looks good, except for a few comments.

Also, when the user clicks send, is there a verification screen for the tx?

**Comment by AnthonyLaw** [comment]
> Also, when the user clicks send, is there a verification screen for the tx?

Yes, this will be another PR.
The final pop-up confirmation will come from metamask.

File: snap/frontend/components/TransferModalBox/index.jsx (hunk 2 above)
Comment (AnthonyLaw) [reply snap/frontend/components/TransferModalBox/index.jsx:44]: good catch! 👍🏼

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**